### DISCUSSION TITLE
Highlights / djvu etc

### DISCUSSION THREAD

**Discussion by meteorquake** [sumatrapdfreader/sumatrapdf#4150] [Ideas]
Quick thought, the annotation tool is quite handy; but it doesn't work on DJVU etc.
Now without needing for any highlights to be saved back to file, is the code adaptable so that highlights can still be made on DJVU or other formats, just not saved?
Cheers, David

**Comment by GitHubRulesOK** [comment]
There was a MOST formats experimental system for some time, TeXT DjVu ?? Ebook ePub PDF  (Last formal VIEWER release  was 3.2) however there were problems with different unit systems 72/96 etc that were difficult to allow for. So fairly soon after expanding the PDF system to be fully compatible with Adobe objects, the other formats support had to be dropped. 

In theory you can write a highlight line in a support file and view with old 3.2 but it would need its own standalone editor (using DjVu compatible units). It has been so so long since opening a DjVu I cant even remember if I have any examples (certainly not on this device)

There is an XPS sample (manual.zip) in https://github.com/sumatrapdfreader/sumatrapdf/issues/1304 but you need 3.1.2 or 3.2 to view it.

Ok dug out one of the problem files and adapted the XPS example to cover the DjVu and now remember some of the old issues . So the file has problems viewing right way up that is a long term open issue with DjVu files and probably for that format there are better rotated page viewers but using SMX with DjVu there are no colours all highlights because the file is greyscale the highlights were too !

Here I am trying to edit the highlight positions in NotePad and it is both slow and not easy, it would be best for many reasons to convert the DjVu to PDF and use PDF highlights.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/3c479edd-bdf7-45d6-81a8-14b94b0a2b67)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/a7d4e734-8ccb-4a91-a387-6793c74bec65)

**Comment by kjk** [comment]
That doesn't seem all that useful and will confuse people and they'll inevitably ask for saving the annotations.

It's possible to add annotations to other formats and save them to external files. In fact, we had experimental code to that but I removed it.

I don't have bandwidth to implement it but if someone wants to try, I can provide guidance.
